pkg/acsengine/orchestrators.go

```diff
@@ -0,0 +1,207 @@
+		// get info for all supported versions
+		for rel, ver := range common.KubeReleaseToVersion {
+			orchs = append(orchs,
+				&api.OrchestratorInfo{
+					Orchestrator: api.Kubernetes,
+					VersionInfo: api.VersionInfo{
+						Release: rel,
+						Version: ver,
+					},
+					Default:    rel == common.KubernetesDefaultRelease,
+					Upgradable: kubernetesUpgrades(rel),
+				})
+		}
+	} else {
+		// get info for the specified release
+		ver, ok := common.KubeReleaseToVersion[release]
+		if !ok {
+			return nil, fmt.Errorf("Kubernetes release %s is not supported", release)
+		}
+		orchs = append(orchs,
+			&api.OrchestratorInfo{
+				Orchestrator: api.Kubernetes,
+				VersionInfo: api.VersionInfo{
+					Release: release,
+					Version: ver,
+				},
+				Default:    release == common.KubernetesDefaultRelease,
+				Upgradable: kubernetesUpgrades(release),
+			})
+	}
+	return orchs, nil
+}
+
+func kubernetesUpgrades(release string) []*api.VersionInfo {
+	ret := []*api.VersionInfo{}
+	switch release {
+	case common.KubernetesRelease1Dot5:
+		ret = append(ret, &api.VersionInfo{Release: common.KubernetesRelease1Dot6, Version: common.KubeReleaseToVersion[common.KubernetesRelease1Dot6]})
+	case common.KubernetesRelease1Dot6:
+		ret = append(ret, &api.VersionInfo{Release: common.KubernetesRelease1Dot6, Version: common.KubeReleaseToVersion[common.KubernetesRelease1Dot6]})
+		ret = append(ret, &api.VersionInfo{Release: common.KubernetesRelease1Dot7, Version: common.KubeReleaseToVersion[common.KubernetesRelease1Dot7]})
+	case common.KubernetesRelease1Dot7:
+		ret = append(ret, &api.VersionInfo{Release: common.KubernetesRelease1Dot7, Version: common.KubeReleaseToVersion[common.KubernetesRelease1Dot7]})
+	}
+	return ret
+}
+
+func dcosInfo(release string) ([]*api.OrchestratorInfo, error) {
+	orchs := []*api.OrchestratorInfo{}
+	if len(release) == 0 {
+		// get info for all supported versions
+		for rel, ver := range common.DCOSReleaseToVersion {
+			orchs = append(orchs,
+				&api.OrchestratorInfo{
+					Orchestrator: api.DCOS,
+					VersionInfo: api.VersionInfo{
+						Release: rel,
+						Version: ver,
+					},
+					Default: rel == common.DCOSDefaultRelease,
+				})
+		}
+	} else {
+		// get info for the specified release
+		ver, ok := common.DCOSReleaseToVersion[release]
+		if !ok {
+			return nil, fmt.Errorf("DCOS release %s is not supported", release)
+		}
+		orchs = append(orchs,
+			&api.OrchestratorInfo{
+				Orchestrator: api.DCOS,
+				VersionInfo: api.VersionInfo{
+					Release: release,
+					Version: ver,
+				},
+				Default: release == common.DCOSDefaultRelease,
+			})
+	}
+	return orchs, nil
+}
+
+func swarmInfo(release string) ([]*api.OrchestratorInfo, error) {
+	return []*api.OrchestratorInfo{
+		{
+			Orchestrator: api.Swarm,
+			VersionInfo: api.VersionInfo{
+				Version: SwarmVersion,
+			},
+			DockerComposeVersion: SwarmDockerComposeVersion,
+		},
+	}, nil
+}
+
+func dockerceInfo(release string) ([]*api.OrchestratorInfo, error) {
```

Comment:
does it not need `Default`?

Reply:
and swarmInfo above

Reply:
no, we support only one version.

Reply:
unless that should make it default ... :)